src/file-upload/handler.ts

```diff
@@ -0,0 +1,77 @@
+import { APIGatewayProxyHandler } from "aws-lambda";
+import { PutObjectCommand, S3Client } from "@aws-sdk/client-s3";
+import { getSignedUrl } from "@aws-sdk/s3-request-presigner";
+
+type CreatePresignedUrlWithClientParams = {
+  region: string;
+  bucket: string;
+  key: string;
+  contentType: string;
+};
+
+/**
+ * Create a presigned URL for a PUT request to upload a file to the specified Amazon S3 bucket
+ *
+ * @param params {CreatePresignedUrlWithClientParams}
+ * @returns A promise that resolves to the presigned URL
+ */
+const createPresignedUrlWithClient = ({ region, bucket, key, contentType }: CreatePresignedUrlWithClientParams) => {
+  const client = new S3Client({ region: region });
+  const command = new PutObjectCommand({
+    Bucket: bucket,
+    Key: key,
+    ContentType: contentType,
+   });
+
+  return getSignedUrl(client, command, { expiresIn: 3600 });
+};
+
+/**
+ * We need to respond with adequate CORS headers.
+ */
+const headers = {
+  "Access-Control-Allow-Origin": "*",
+  "Access-Control-Allow-Credentials": true
+};
+
+/**
+ * Lambda function that returns a presigned URL for a PUT request to upload a file to the specified Amazon S3 bucket
+ */
+export const main: APIGatewayProxyHandler = async ({ body }) => {
+  try {
+    const { path, contentType } = JSON.parse(body || '{}');
+    const { BUCKET_NAME, BUCKET_REGION } = process.env;
+
+    if (!path) {
+      throw Error("Invalid request body");
```

Comment:
Could it be worth setting a status code for the catch here?

Reply:
Resolved.